packages/amplify-category-api/src/index.ts

```diff
@@ -154,43 +148,28 @@ export const initEnv = async (context: $TSContext): Promise<void> => {
   }
 
   /**
-   * Check team provider info to ensure it hasn't already been created for current env
+   * Check environment parameter manager to ensure it hasn't already been created for current env
    */
-  const currentEnv = amplify.getEnvInfo().envName;
-  const teamProviderInfo = stateManager.getTeamProviderInfo();
-  if (
-    teamProviderInfo[currentEnv][categories]
-    && teamProviderInfo[currentEnv][categories][category]
-    && teamProviderInfo[currentEnv][categories][category][resourceName]
-    && teamProviderInfo[currentEnv][categories][category][resourceName]
-    && teamProviderInfo[currentEnv][categories][category][resourceName][rdsRegion]
-  ) {
+  const envParamManager = (await ensureEnvParamManager()).instance;
+  if (envParamManager.hasResourceParamManager(category, resourceName)) {
```

Comment:
The only difference I see from the previous case is that we were checking for existence of `rdsRegion` parameter. Is it possible to add that check?